crates/lox-time/src/offsets.rs

```diff
@@ -26,33 +21,43 @@ where
     ) -> Result<TimeDelta, Self::Error>;
 }
 
-#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
-pub struct DefaultOffsetProvider;
+pub trait Offset<Origin, Target>: OffsetProvider
+where
+    Origin: TimeScale,
+    Target: TimeScale,
+{
+    fn offset(&self, origin: Origin, target: Target, delta: TimeDelta) -> TimeDelta;
+}
 
-impl<Origin, Target> TryOffset<Origin, Target> for DefaultOffsetProvider
+impl<T, Origin, Target> Offset<Origin, Target> for T
 where
     Origin: TimeScale,
     Target: TimeScale,
-    DefaultOffsetProvider: Offset<Origin, Target>,
+    T: TryOffset<Origin, Target, Error = Infallible>,
```

Comment:
Found a way to have my cake and also eat it (with minimal caveats). I am defining the `Offset` trait in terms of `TryOffset` instead of the other way around. This does require some funky `From` impls for errors that will never be triggered (see below). These will go away once `!` is landing in stable,